### PR TITLE
Update llvm commit ID to 1d01fc1

### PR DESCRIPTION
**Pull request by imaihal**
This PR includes fixes for LLVM revisions: [D85495](https://reviews.llvm.org/D85495), [D85622](https://reviews.llvm.org/D85622), and [D86121](https://reviews.llvm.org/D86121) to use newer LLVM commit.

- D85495
`TypeID` should be set to `mlir::Dialect` by LLVM revision [D85495](https://reviews.llvm.org/D85495) I updated as it is done at `ToyDialect` in the revision https://reviews.llvm.org/D85495#change-M9kXqI8sOFkn

- D85622
This revision includes change in how to register dialects. I updated as it is done in the revision. For example. https://reviews.llvm.org/D85622#change-2drP7BXp5CDr

- D86121
`type.getKind()` was removed by LLVM revision [D86121](https://reviews.llvm.org/D86121). So, I needed to modify it as described in https://llvm.discourse.group/t/removing-kinds-from-attributes-and-types/1547
I used `TypeSwitch<Type>(type)` instead of `switch (type.getKind())` 

I confirmed this works on [1d01fc1](https://github.com/llvm/llvm-project/commit/1d01fc100bb5bef5f5eaf92520b2e52f64ee1d6e) (Aug 31, 2020)

### CODE REVIEW

**Comment by AlexandreEichenberger** [comment]
Thanks for updating the LLVM level. In addition to the usual updates to the commit level, I see a lot of work on types. Can you comment on whether that is due to upgrades in MLIR or separate changes.

**Comment by imaihal** [comment]
Thanks for your comments. All of the work in this PR is due to upgrade MLIR.

Regarding the types,  `type.getKind()` was removed by LLVM revision [D86121](https://reviews.llvm.org/D86121). So, I needed to modify it as described in https://llvm.discourse.group/t/removing-kinds-from-attributes-and-types/1547
I used `TypeSwitch<Type>(type)` instead of `switch (type.getKind())` 

I added descriptions about other works above.

**Comment by imaihal** [comment]
I'm not sure how to fix other errors in travis and docker.  I guess this errors come from using old llvm images, but I don't know how to fix them. Could you give me your comments or suggestions?

**Comment by AlexandreEichenberger** [comment]
@caoimhinuibrian Since the LLVM level is changed, I believe you mentioned that some of the tests will fail. Can you just check the list to see if the failures are expected? Thanks

**Comment by caoimhinuibrian** [comment]
@AlexandreEichenberger @imaihal When the LLVM commit level changes, the TravisCI builds will always fail. The prereq docker images are for the previous version of LLVM. If you have verified that the build works in your fork, then it should be fine to merge the PR and subsequently update the prereq images for 390x and ppc64le. If you can't do that yourself, ask me to do it.

File: src/Dialect/Krnl/KrnlOps.hpp [hunk 1]
```diff
@@ -38,11 +39,17 @@ class KrnlOpsDialect : public Dialect {
 
   /// Print a type registered to this dialect.
   void printType(Type type, DialectAsmPrinter &os) const override {
+    TypeSwitch<Type>(type).Case<LoopType>([&](Type) {
+      os << "loop";
+      return;
+    });
+    /*
```
Comment (AlexandreEichenberger) [inline src/Dialect/Krnl/KrnlOps.hpp:46]: Please remove the old commented code

Comment (imaihal) [reply src/Dialect/Krnl/KrnlOps.hpp]: I removed this code.

**Comment by imaihal** [comment]
I confirmed FileCheck test and backend test passed on my s390 environment. However, `linux-s390x-docker` of this PR  fails in addition to `travis-ci`. Is this expected results?  If so, could you merge this?

@caoimhinuibrian I don't know how to create the prereq images. Could you create them?

**Comment by tjingrant** [comment]
@imaihal I'm dealing with building pre-req images now, I will base my commits on this patch for convenience.

**Comment by AlexandreEichenberger** [comment]
@imaihal could you update and try the test again? there was a problem with the linux-s390x and Gong did some work to fix it, Thanks

**Comment by tjingrant** [comment]
Closing in favor of https://github.com/onnx/onnx-mlir/pull/292, we need privileged information on Travis to push new prereq docker images so it has to be a branch in the master repo, not a pull request from a forked repo.

**Pull request closed without merging**